tracing/tracing.go

```diff
@@ -6,51 +6,204 @@ package tracing
-	if err != nil {
-		return nil, errors.Wrap(err, "could not initialize jaeger tracer")
-	}
-	return closer, nil
-}
+const (
+	envSamplerParam           = "JAEGER_SAMPLER_PARAM"
+	envEndpoint               = "JAEGER_ENDPOINT"
+	envJaegerAgentHost        = "JAEGER_AGENT_HOST"
+	envJaegerAgentPort        = "JAEGER_AGENT_PORT"
+	envSamplerType            = "JAEGER_SAMPLER_TYPE"
+	envSamplerRemoteURL       = "JAEGER_SAMPLING_ENDPOINT"
+	DefaultSamplingServerPort = 5778
+)
 
 // NewFromEnv is a convenience function to allow tracing configuration
 // via environment variables
 //
 // Tracing will be enabled if one (or more) of the following environment variables is used to configure trace reporting:
 // - JAEGER_AGENT_HOST
-// - JAEGER_SAMPLER_MANAGER_HOST_PORT
-func NewFromEnv(serviceName string, options ...jaegercfg.Option) (io.Closer, error) {
-	cfg, err := jaegercfg.FromEnv()
+// - JAEGER_ENDPOINT
+func NewFromEnv(serviceName string) (io.Closer, error) {
+	cfg, err := parseTracingConfig()
 	if err != nil {
 		return nil, errors.Wrap(err, "could not load jaeger tracer configuration")
 	}
-
-	if cfg.Sampler.SamplingServerURL == "" && cfg.Reporter.LocalAgentHostPort == "" && cfg.Reporter.CollectorEndpoint == "" {
+	if cfg.samplerRemoteURL == "" && cfg.host == "" && cfg.collectorURL == "" {
 		return nil, ErrBlankTraceConfiguration
 	}
+	return cfg.initJaegerTracerProvider(serviceName)
+}
+
+// parseAttributes Parse Jaeger tags from env var JAEGER_TAGS, example of TAGs format: key1=value1,key2=${value2:value3} where value2 is an env var
+// and value3 is the default value, which is optional.
+func parseAttributes(sTags string) ([]attribute.KeyValue, error) {
+	pairs := strings.Split(sTags, ",")
+	res := make([]attribute.KeyValue, 0, len(pairs))
+	for _, p := range pairs {
+		k, v, found := strings.Cut(p, "=")
+		if found {
+			k, v := strings.TrimSpace(k), strings.TrimSpace(v)
+			if strings.HasPrefix(v, "${") && strings.HasSuffix(v, "}") {
+				e, d, _ := strings.Cut(v[2:len(v)-1], ":")
+				v = os.Getenv(e)
+				if v == "" && d != "" {
+					v = d
+				}
+			}
+			if v == "" {
+				return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+			}
+			res = append(res, attribute.String(k, v))
+		} else if p != "" {
+			return nil, errors.Errorf("invalid tag %q, expected key=value", p)
+		}
+	}
+	return res, nil
+}
+
+type config struct {
+	host             string
+	collectorURL     string
+	port             int
+	samplerType      string
+	samplerRemoteURL string
+	samplerParam     float64
+	customAttributes []attribute.KeyValue
+}
+
+// parseTracingConfig facilitates initialization that is compatible with Jaeger's InitGlobalTracer method.
+func parseTracingConfig() (config, error) {
+	cfg := config{}
+	var err error
+
+	// We first parse the collector endpoint, if it is not exist then we parse the agent endpoint, if it is not exist, we default to localhost:6831
+	if e := os.Getenv(envEndpoint); e != "" {
```

Comment:
I was expecting `parseTracingConfig()` to be close to `samplerConfigFromEnv()` from `github.com/uber/jaeger-client-go/config/config_env.go`, but in this function the sampling server URL setup logic looks more complicated. Can you just help me understand why we need a different logic here, please?